src/neuromaps_nhp/graph.py

```diff
@@ -0,0 +1,394 @@
+"""Module for managing brain template surface spaces and their transformations."""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import matplotlib.cm as cm
+import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
+import networkx as nx
+import numpy as np
+import yaml
+from matplotlib.lines import Line2D
+
+
+@dataclass
+class Resource(ABC):
+    """Abstract base class for brain template resources."""
+
+    name: str
+    description: str
+    file_path: Path
+
+    @abstractmethod
+    def fetch(self) -> Path:
+        """Fetch the resource."""
+        pass
+
+
+@dataclass
+class Surface_Atlas(Resource):
+    """Surface atlas for brain template spaces."""
+
+    space: str
+    density: str
+    hemisphere: str
+    resource_type: str
+
+    def fetch(self) -> Path:
+        """Fetch the surface resource."""
+        return self.file_path
+
+
+@dataclass
+class Surface_Transform(Resource):
+    """Transform resource between brain template spaces."""
+
+    source_space: str
+    target_space: str
+    density: str
+    hemisphere: str
+    resource_type: str
+
+    def fetch(self) -> Path:
+        """Fetch the transform resource."""
+        return self.file_path
+
+
+class SurfaceGraph:
+    """Multi-directed graph of brain template spaces and their transformations."""
+
+    def __init__(self) -> None:
+        """Initialize an empty SurfaceGraph."""
+        self._spaces: Dict[str, List[Surface_Atlas]] = {}
+        self.graph = nx.MultiDiGraph()  # MultiDiGraph for multiple edges between nodes
+
+    def add_space(self, space_obj: Surface_Atlas) -> None:
+        """Add a space node to the graph."""
+        space_name = space_obj.space
+        if space_name not in self._spaces:
+            self._spaces[space_name] = []
+        self._spaces[space_name].append(space_obj)
```

Comment:
Here surfaces are stored in `self._spaces` below they are duplicated in attributes. Pick one